src/internal/mod.rs

```diff
@@ -334,6 +334,12 @@ impl From<PublicKey> for crate::proto::transform::PublicKey {
         proto_pub_key
     }
 }
+impl TryFrom<&[u8]> for PublicKey {
+    type Error = IronOxideErr;
+    fn try_from(key_bytes: &[u8]) -> Result<PublicKey, IronOxideErr> {
+        PublicKey::new_from_slice(key_bytes.split_at(RecryptPublicKey::ENCODED_SIZE_BYTES / 2))
```

Comment:
This will panic if key_bytes is shorter than the split. We should check that ahead of time.

Reply:
- Added a check that `key_bytes` has a length of `RecryptPublicKey::ENCODED_SIZE_BYTES`. If it is not, it will give `IronOxideErr::WrongSizeError`.
- Added a test with a key_bytes of length 8.